workers/proof-manager/src/mock.rs

```diff
@@ -31,134 +43,121 @@ pub struct MockProofManager;
-        Handle::current().spawn_blocking(move || Self::execution_loop(&job_queue));
+        Handle::current().spawn_blocking(move || {
+            if let Err(e) = Self::execution_loop(&job_queue) {
+                log::error!("error in mock proof manager: {e}");
+            }
+        });
     }
 
     /// The execution loop for the mock
-    fn execution_loop(job_queue: &Receiver<ProofManagerJob>) {
+    fn execution_loop(job_queue: &Receiver<ProofManagerJob>) -> Result<(), ProofManagerError> {
         loop {
             match job_queue.recv() {
                 Err(e) => {
                     log::error!("error receiving job: {e}... shutting down");
-                    break;
+                    return Err(ProofManagerError::JobQueueClosed("job queue closed".to_string()));
                 },
-                Ok(job) => Self::handle_job(job.type_, job.response_channel),
+                Ok(job) => Self::handle_job(job.type_, job.response_channel)?,
             }
         }
     }
 
     /// Handle a job by immediately returning a dummy proof
-    fn handle_job(job_type: ProofJob, response_channel: TokioSender<ProofBundle>) {
+    fn handle_job(
+        job_type: ProofJob,
+        response_channel: TokioSender<ProofBundle>,
+    ) -> Result<(), ProofManagerError> {
         let bundle = match job_type {
             ProofJob::ValidWalletCreate { witness, statement } => {
-                ProofBundle::ValidWalletCreate(Self::valid_wallet_create(&witness, statement))
+                ProofBundle::ValidWalletCreate(Self::valid_wallet_create(witness, statement)?)
             },
             ProofJob::ValidWalletUpdate { witness, statement } => {
-                ProofBundle::ValidWalletUpdate(Self::valid_wallet_update(&witness, statement))
+                ProofBundle::ValidWalletUpdate(Self::valid_wallet_update(witness, statement)?)
             },
             ProofJob::ValidReblind { witness, statement } => {
-                ProofBundle::ValidReblind(Self::valid_reblind(&witness, statement))
+                ProofBundle::ValidReblind(Self::valid_reblind(witness, statement)?)
             },
             ProofJob::ValidCommitments { witness, statement } => {
-                ProofBundle::ValidCommitments(Self::valid_commitments(&witness, statement))
+                ProofBundle::ValidCommitments(Self::valid_commitments(witness, statement)?)
             },
             ProofJob::ValidMatchSettleSingleprover { witness, statement } => {
-                ProofBundle::ValidMatchSettle(Self::valid_match_settle(&witness, statement))
+                ProofBundle::ValidMatchSettle(Self::valid_match_settle(witness, statement)?)
             },
         };
 
         response_channel.send(bundle).expect(ERR_RESPONSE_CHANNEL_CLOSED);
+        Ok(())
     }
 
     /// Generate a dummy proof of `VALID WALLET CREATE`
     fn valid_wallet_create(
-        _witness: &SizedValidWalletCreateWitness,
+        witness: SizedValidWalletCreateWitness,
```

Comment:
clippy complaining about pass by value here, and for the other mock proof generation methods